software/metax/genotype/CYVCF2Genotype.py

```diff
@@ -40,12 +40,12 @@ def vcf_file_geno_lines(path, mode="genotyped", variant_mapping=None, whitelist=
                 for sample in variant.genotypes:
                     d_ = (sample[0] == a+1) + (sample[1] == a+1)
                     d.append(d_)
-                f = numpy.mean(numpy.array(d,dtype=numpy.int32))/2
+                f = numpy.nanmean(numpy.array(d,dtype=numpy.int32))/2
                 yield (variant_id, chr, pos, ref, alt, f) + tuple(d)
 
         elif mode == "imputed":
-            if len(alts) > 1:
-                logging.log("VCF imputed mode doesn't support multiple ALTs, skipping %s", variant_id)
+            if (len(ref)) | (len(alts[0])) > 1:
```

Comment:
The genotype contains indels too?

Reply:
Yes, and they could be coded as either REF or ALT.